src/expr/src/relation/func.rs

```diff
@@ -580,10 +580,39 @@ fn lag_lead_inner_respect_nulls<'a>(
     result
 }
 
+#[allow(clippy::as_conversions)]
 fn lag_lead_inner_ignore_nulls<'a>(
     args: Vec<(Datum<'a>, Datum<'a>, Datum<'a>)>,
     lag_lead_type: &LagLeadType,
 ) -> Vec<Datum<'a>> {
+    if i64::try_from(args.len()).is_err() {
+        panic!("window partition way too big")
+    }
+    // Preparation: Make sure we can jump over a run of nulls in constant time, i.e., regardless of
+    // how many nulls the run has. The following skip tables will point to the next non-null index.
+    let mut skip_nulls_backward = Vec::new();
+    let mut last_non_null: i64 = -1;
+    for (i, (d, _, _)) in args.iter().enumerate() {
+        if d.is_null() {
+            skip_nulls_backward.push(Some(i64::cast_from(last_non_null)));
+        } else {
+            skip_nulls_backward.push(None);
+            last_non_null = i as i64;
```

Comment:
Same for this cast.

Reply:
This one is needed because `i` is an `usize` due to the `enumerate()`.